src/core/iam/nodeinfoprovider/nodeinfoprovider.cpp

```diff
@@ -7,7 +7,22 @@
 
 #include "nodeinfoprovider.hpp"
 
-static bool CaseInsensitiveEqual(const aos::String& lhs, const aos::String& rhs)
+namespace aos::iam::nodeinfoprovider {
+
+namespace {
+
+/***********************************************************************************************************************
+ * Consts
+ **********************************************************************************************************************/
+
+constexpr auto cNodeComponentStrLen  = 8;
+constexpr auto cMaxNumNodeComponents = static_cast<size_t>(CoreComponentEnum::eNumComponents);
+
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+bool CaseInsensitiveEqual(const aos::String& lhs, const aos::String& rhs)
```

Comment:
As I remember we discussed to move this function into tools/string.hpp as `Compare` method with case sensitivity parameter.